File: FrozenApp/src/main/java/com/github/lakeshire/frozen/SubListFragment.java
package com.github.lakeshire.frozen;

import android.view.View;
import android.widget.AdapterView;

import com.github.lakeshire.frozen.manager.SubjectAdapter;

import java.util.ArrayList;
import java.util.List;

import lakeshire.github.com.frozenframework.adapter.BaseAdapter;
import lakeshire.github.com.frozenframework.fragment.BaseListFragment;
import lakeshire.github.com.frozenframework.fragment.IPager;
import lakeshire.github.com.frozenframework.manager.PopupWindowManager;
import lakeshire.github.com.frozenframework.model.MenuItem;
import lakeshire.github.com.frozenframework.util.CustomToast;
import lakeshire.github.com.frozenframework.util.RandomUtil;

/**
 * Created by louis.liu on 2017/9/6.
 */

public class SubListFragment extends BaseListFragment<String> implements IPager {
    private String tabTitle;

    @Override
    protected void initData() {
//        for (int i = 0; i < 10; i++) {
//            mDataList.add(tabTitle + RandomUtil.getRandom(1000));
//        }
        showLoadingLayout();
    }

    @Override
    protected BaseAdapter<String> getAdapter() {
        return new SubjectAdapter(getContext(), mDataList, R.layout.item);
    }

    @Override
    protected void loadMoreData() {
        for (int i = 0; i < 10; i++) {
            mDataList.add(tabTitle + RandomUtil.getRandom(1000));
        }
        notifyAdapter(STATUES_OK, false);
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        startFragment(SubFragment.class, null);
//        PopupWindowManager.pop(getActivity(), view);
    }

    @Override
    public void initUi() {
        super.initUi();

        List<MenuItem> items = new ArrayList<>();
        items.add(new MenuItem("试听", R.drawable.ic_category, null));
        items.add(new MenuItem("下载", R.drawable.ic_home, null));
        PopupWindowManager.initPopupWindow(getActivity(), items, new AdapterView.OnItemClickListener() {


            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                switch (position) {
                    case 0:
                        new CustomToast(getContext()).showToast("试听");
                        break;
                    case 1:
                        new CustomToast(getContext()).showToast("下载");
                        break;
                }
            }
        });
    }


    @Override
    public void onRefresh() {
        mDataList.clear();
        for (int i = 0; i < 10; i++) {
            mDataList.add(tabTitle + RandomUtil.getRandom(1000));
        }
        notifyAdapter(STATUES_OK, true);
    }

    @Override
    public String getTabTitle() {
        return this.tabTitle;
    }

    @Override
    public void setTabTitle(String title) {
        this.tabTitle = title;
    }
}

File: FrozenApp/src/main/java/com/github/lakeshire/frozen/SubListFragment.kt
package com.github.lakeshire.frozen

import android.view.View
import android.widget.AdapterView
import com.github.lakeshire.frozen.manager.SubjectAdapter
import lakeshire.github.com.frozenframework.adapter.BaseAdapter
import lakeshire.github.com.frozenframework.fragment.BaseListFragment
import lakeshire.github.com.frozenframework.fragment.IPager
import lakeshire.github.com.frozenframework.manager.PopupWindowManager
import lakeshire.github.com.frozenframework.model.MenuItem
import lakeshire.github.com.frozenframework.util.CustomToast
import lakeshire.github.com.frozenframework.util.RandomUtil
import java.util.*

class SubListFragment : BaseListFragment<String>(), IPager {
    override var tabTitle: String = ""

    override fun initData() {
        //        for (int i = 0; i < 10; i++) {
        //            mDataList.add(tabTitle + RandomUtil.getRandom(1000));
        //        }
        showLoadingLayout()
    }

    override val adapter: BaseAdapter<String>
        get() = SubjectAdapter(context, mDataList, R.layout.item)

    override fun loadMoreData() {
        for (i in 0..9) {
            mDataList.add(tabTitle!! + RandomUtil.getRandom(1000))
        }
        notifyAdapter(BaseListFragment.STATUES_OK, false)
    }

    override fun onItemClick(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        startFragment(SubFragment::class.java, null)
        //        PopupWindowManager.pop(getActivity(), view);
    }

    override fun initUi() {
        super.initUi()

        val items = ArrayList<MenuItem>()
        items.add(MenuItem("试听", R.drawable.ic_category, null))
        items.add(MenuItem("下载", R.drawable.ic_home, null))
        PopupWindowManager.initPopupWindow(activity, items) { parent, view, position, id ->
            when (position) {
                0 -> CustomToast(context).showToast("试听")
                1 -> CustomToast(context).showToast("下载")
            }
        }
    }


    override fun onRefresh() {
        mDataList.clear()
        for (i in 0..9) {
            mDataList.add(tabTitle!! + RandomUtil.getRandom(1000))
        }
        notifyAdapter(BaseListFragment.STATUES_OK, true)
    }
}
